tests/Aspire.Hosting.PostgreSQL.Tests/PostgresFunctionalTests.cs

```diff
@@ -210,31 +210,24 @@ await pipeline.ExecuteAsync(async token =>
         }, cts.Token);
     }
 
-    [Fact(Skip = "https://github.com/dotnet/aspire/issues/5325")]
+    [Fact]
     [RequiresDocker]
     public async Task VerifyWithPgWeb()
     {
         var cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));
 
         using var builder = TestDistributedApplicationBuilder.CreateWithTestContainerRegistry(testOutputHelper);
 
+        IResourceBuilder<PgWebContainerResource>? pgWebBuilder = null;
         var dbName = "postgres";
-        var pg = builder.AddPostgres("pg1").WithPgWeb().AddDatabase(dbName);
-
-        builder.Services.AddHttpClient();
+        var pg = builder.AddPostgres("pg1").WithPgWeb(c=> pgWebBuilder = c).AddDatabase(dbName);
+        Assert.NotNull(pgWebBuilder);
 
         using var app = builder.Build();
 
         await app.StartAsync();
 
-        var factory = app.Services.GetRequiredService<IHttpClientFactory>();
-        var client = factory.CreateClient();
-
-        var pgWeb = builder.Resources.OfType<PgWebContainerResource>().SingleOrDefault();
-
-        var pgWebEndpoint = pgWeb!.PrimaryEndpoint;
-
-        var testConnectionApiUrl = $"{pgWebEndpoint.Scheme}://{pgWebEndpoint.Host}:{pgWebEndpoint.Port}/api/connect";
+        var client = app.CreateHttpClient(pgWebBuilder.Resource.Name, "http");
 
         var pipeline = new ResiliencePipelineBuilder().AddRetry(new Polly.Retry.RetryStrategyOptions
```

Comment:
Pgadmin looks for a log to see if it's ready. Can we do something similar here for pgweb?

Reply:
Can be a follow up

Reply:
#5752